app/src/main/java/ch/epfl/polybazaar/listing/ListingDatabase.java

```diff
@@ -41,7 +46,13 @@ public static void fetchListing(final String listingID, final ListingCallback ca
      * @param callback a SuccessCallback interface implementation
      */
     public static void storeListing(final Listing listing, final String listingID, final SuccessCallback callback){
-        setData(listingCollectionName, listingID, listing, callback);
+        db = DatastoreFactory.getDependency();
+        Map<String,Object> data = new HashMap<>();
+        data.put("title",listing.getTitle());
+        data.put("description",listing.getDescription());
+        data.put("price",listing.getPrice());
+        data.put("userEmail",listing.getUserEmail());

```

Comment:
This is a bad idea, we'll have to change this every time we change the User class. Try to simply store the User object directly. Firebase will resolve the field names. This also requires to change the DataSnapshot class. Same goes for Listing and LiteListing.